query-engine/connectors/sql-query-connector/src/query_builder/read.rs

```diff
@@ -124,9 +124,7 @@ where
     T: SelectDefinition,
 {
     let (select, additional_selection_set) = query.into_select(model, aggr_selections, ctx);
-    let select = columns
-        .map(|c| c.set_is_selected(true))
-        .fold(select, |acc, col| acc.column(col));
+    let select = columns.fold(select, |acc, col| acc.column(col));
```

Comment:
get_records is no longer responsible for setting the hacky `is_selected` field, the caller is

Reply:
This makes it possible in `aggregate` and `group_by` to not mark columns of the sub-select as selected and thus not cast them.

Here's an example:

```sql
SELECT
  MAX(col) -- mark as selected
FROM (
  SELECT col FROM "Table" -- not mark as selected
) as "sub"
```